src/kwlib.sh

```diff
@@ -406,3 +406,35 @@ function generate_tarball()
     exit $ret
   fi
 }
+
+# This function extracts a .tar.gz file to a given path
+#
+# @file_to_extract The path to the file to extract from
+# @path Where to extract the file
+# @compression_type compression program used
+# @flag How to display (or not) the command used
+function extract_tarball()
+{
+  local file_to_extract="$1"
+  local path="$2"
+  local compression_type="$3"
```

Comment:
I'm not sure, but I think the option `-a` from tar command is an automatic mechanism to check the compression type and use it (see `-taf`). Let's use this option as a default.

Reply:
Did you check the `-a` option? Can we use it as the default?

Reply:
Yes, it's the same as `--auto-compress`, and I opted to set it as default a few lines below (line 457)

Reply:
Nice, I missed that. Thanks.